### PR TITLE
DM-7827:add file group search processor

### PR DESCRIPTION
**Pull request by lznakano**
Please test it while I am away.  To test it,
  1. check out this branch
  2. start vpn 
  3. select LSST Data/Images
  4. enter 9.469,-1.152
  5. highlight one or more columns and then select "Prepare Download"
  
The current external file name conversion is
  DeepCoadd/deepCoaddId-filter.fits
  ScienceCCD/scienceCcdId-filter.fits

Thanks!

### CODE REVIEW

**Comment by ymeiymei** [comment]
Tested both deep co-adds and the science ccd exposure images, with single image or multiple images, with folder or without folder. All worked great! I also loaded the downloaded images back to firefly and ds9. They looked good to me.

File: src/firefly/java/edu/caltech/ipac/firefly/server/query/lsst/LSSTFileGroupProcessor.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+/**
+ * Created by zhang on 11/17/16.
+ *
+ */
+
+@SearchProcessorImpl(id = "LSSTFileGroupProcessor")
+public class LSSTFileGroupProcessor  extends FileGroupsProcessor {
+    public static final String LSST_FILESYSTEM_BASEPATH = AppProperties.getProperty("lsst.filesystem_basepath");
+    private  Logger.LoggerImpl logger = getLogger();
+    private static  long  SIZE = 30611520;
```
Comment (loitly) [inline src/firefly/java/edu/caltech/ipac/firefly/server/query/lsst/LSSTFileGroupProcessor.java]: Is this size true for both cases; deep coadd and science ccd?

File: src/firefly/java/edu/caltech/ipac/firefly/server/query/lsst/LSSTFileGroupProcessor.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        String zipType = request.getParam("zipType");
+        boolean zipFolders = true;
+        if (zipType != null && zipType.equalsIgnoreCase("flat")) {
+            zipFolders = false;
+        }
+
+        //String basePath = LSST_FILESYSTEM_BASEPATH;
+        //basePath = basePath!=null? basePath :"";
+        String baseFileName = request.getParam(DownloadRequest.BASE_FILE_NAME);
+        String basePath  = baseFileName ;
```
Comment (loitly) [inline src/firefly/java/edu/caltech/ipac/firefly/server/query/lsst/LSSTFileGroupProcessor.java]: Should clean up unused variables, ie.  fgSize, basePath.

File: src/firefly/java/edu/caltech/ipac/firefly/server/query/lsst/LSSTFileGroupProcessor.java [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+        else{
+            Long run =  (Long) dgData.get(rowIdx,"run");
+            Integer camcol =  (Integer)dgData.get(rowIdx,"camcol");
+            Long field =  (Long) dgData.get(rowIdx,"field");
+            String filterName =  (String) dgData.get(rowIdx,"filterName");
+
+            svr.setParam("run", run.toString());
+            svr.setParam("camcol", camcol.toString());
+            svr.setParam("field", field.toString());
+            svr.setParam("filterName", filterName);
```
Comment (loitly) [inline src/firefly/java/edu/caltech/ipac/firefly/server/query/lsst/LSSTFileGroupProcessor.java]: svr was created and then populated but never used.  why?

File: src/firefly/java/edu/caltech/ipac/firefly/server/query/lsst/LSSTImageSearch.java [hunk 4]
```diff
@@ -22,9 +20,9 @@
  * Created by zhang on 11/3/16.
  */
 public class LSSTImageSearch extends URLFileInfoProcessor {
-//    private static final Logger.LoggerImpl _log = Logger.getLogger();
+    private static final Logger.LoggerImpl _log = Logger.getLogger();
     private static String DAX_URL="http://lsst-qserv-dax01.ncsa.illinois.edu:5000/image/v0/";
-
+    private static FileInfo fileInfo;
```
Comment (loitly) [inline src/firefly/java/edu/caltech/ipac/firefly/server/query/lsst/LSSTImageSearch.java]: added but, not used.

Comment (lznakano) [inline src/firefly/java/edu/caltech/ipac/firefly/server/query/lsst/LSSTImageSearch.java]: I did not clean up the codes after making changes.  Many variables were used before and were no longer needed after I made the code changes.  I will clean the codes and them merge.
Thanks!

**Pull request merged**